src/Tools.pm

```diff
@@ -694,7 +694,7 @@ sub GetSysconfigValue
 
   if(open my $fh, Bootloader::Path::Sysconfig()) {
     while(<$fh>) {
-      $val = $1, last if /^\s*$key\s*=\s*"(.*)"\s*$/;
+      $val = $2, last if /^\s*$key\s*=\s*(["']?+)(.*)\1\s*$/
```

Comment:
The '+' you omitted actually matters. it's the difference of accepting a line like
```
a = "3
```

or not (do not backtrack on the space after the '=').

Reply:
OTOH, my version is then failing with `Error: Initializing for unknown bootloader  "grub2, fallback to none` instead of `Use of uninitialized value $loader`

Reply:
A good point. It's better to match anything and complain later. So I would go back to your original proposal or something like `=\s*(["']?)(.*?)\1?\s*$` if you want to include also single quotes.